web-admin/src/routes/[organization]/-/users/groups/+page.svelte

```diff
@@ -0,0 +1,308 @@
+    createAdminServiceSetOrganizationMemberUsergroupRole,
+    getAdminServiceListOrganizationMemberUsergroupsQueryKey,
+    getAdminServiceListOrganizationMemberUsersQueryKey,
+    getAdminServiceListUsergroupMemberUsersQueryKey,
+  } from "@rilldata/web-admin/client";
+  import DelayedSpinner from "@rilldata/web-common/features/entity-management/DelayedSpinner.svelte";
+  import OrgGroupsTable from "@rilldata/web-admin/features/organizations/users/OrgGroupsTable.svelte";
+  import { eventBus } from "@rilldata/web-common/lib/event-bus/event-bus";
+  import { useQueryClient } from "@tanstack/svelte-query";
+  import Button from "@rilldata/web-common/components/button/Button.svelte";
+  import { Plus } from "lucide-svelte";
+  import CreateUserGroupDialog from "@rilldata/web-admin/features/organizations/users/CreateUserGroupDialog.svelte";
+  import { Search } from "@rilldata/web-common/components/search";
+
+  let userGroupName = "";
+  let isCreateUserGroupDialogOpen = false;
+  let searchText = "";
+
+  $: organization = $page.params.organization;
+  // FIXME: Ideally, we have membersCount in this response
+  $: listOrganizationMemberUsergroups =
+    createAdminServiceListOrganizationMemberUsergroups(organization);
+  $: listOrganizationMemberUsers =
+    createAdminServiceListOrganizationMemberUsers(organization);
+
+  const queryClient = useQueryClient();
+  const currentUser = createAdminServiceGetCurrentUser();
+  const createUserGroup = createAdminServiceCreateUsergroup();
+  const renameUserGroup = createAdminServiceRenameUsergroup();
+  const deleteUserGroup = createAdminServiceDeleteUsergroup();
+  const addUserGroupRole = createAdminServiceAddOrganizationMemberUsergroup();
+  const setUserGroupRole =
+    createAdminServiceSetOrganizationMemberUsergroupRole();
+  const revokeUserGroupRole =
+    createAdminServiceRemoveOrganizationMemberUsergroup();
+  const removeUserGroupMember = createAdminServiceRemoveUsergroupMemberUser();
+  const addUsergroupMemberUser = createAdminServiceAddUsergroupMemberUser();
+
+  $: filteredGroups = $listOrganizationMemberUsergroups.data?.members.filter(
+    (group) => group.groupName.toLowerCase().includes(searchText.toLowerCase()),
+  );
+
+  async function handleCreate(newName: string) {
+    try {
+      await $createUserGroup.mutateAsync({
+        organization: organization,
+        data: {
+          name: newName,
+        },
+      });
+
+      await queryClient.invalidateQueries(
+        getAdminServiceListOrganizationMemberUsergroupsQueryKey(organization),
+      );
+
+      userGroupName = "";
+      isCreateUserGroupDialogOpen = false;
+
+      eventBus.emit("notification", { message: "User group created" });
+    } catch (error) {
+      console.error("Error creating user group", error);
+      eventBus.emit("notification", {
+        message: "Error creating user group",
+        type: "error",
+      });
+    }
+  }
+
+  async function handleRename(groupName: string, newName: string) {
```

Comment:
Similarly, this (and the other User Group actions in this file) is getting prop-drilled: `+page.svelte` -> `OrgGroupsTable.svelte` -> `OrgGroupsActionCell.svelte` -> `EditUserGroupDialog.svelte`